### PR TITLE
Proposal for Blockchain Explorer for Iroha

### PR DESCRIPTION
**Pull request by turuslan**
Signed-off-by: turuslan <turuslan.devbox@gmail.com>

### CODE REVIEW

**Comment by trbs** [comment]
- Please update the labs short description; Think the first paragraph of the wiki page would be a good addition.
- Please update the sponsor; Afaik "Hyperledger Internship project" does not qualify ;-) 

On more general note and without knowing all the details it seems a shame (and wasted opportunity ?) to fork and make it incompatible... Isn't there a way to work with hyperledger/blockchain-explorer to at least have a path towards a common codebase ?

File: labs/blockchain-explorer-iroha.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Lab Name
+[Integration of Hyperledger Iroha into Hyperledger Explorer tool](https://wiki.hyperledger.org/display/INTERN/Integration+of+Hyperledger+Iroha+into+Hyperledger+Explorer+tool)
```
Comment (tkuhrt) [inline labs/blockchain-explorer-iroha.md]: Lab name should be what we will end up calling the github repository. Maybe something more like iroha-explorer would be appropriate.

File: labs/blockchain-explorer-iroha.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# Scope of Lab
+Code of [hyperledger/blockchain-explorer](https://github.com/hyperledger/blockchain-explorer) supported only Hyperledger Fabric and needed refactoring.  
+This fork supports only Hyperledger Iroha. It rebuilds backend part from scratch and refactors frontend part.
+
+# Initial Committers
+- https://github.com/turuslan
+- https://github.com/Pobepto
+
+# Sponsor
+Hyperledger Internship project
```
Comment (tkuhrt) [inline labs/blockchain-explorer-iroha.md]: Lab sponsor needs to be a member of the TSC, a maintainer of one of the projects, or a working group chair. Please provide a sponsor that meets these criteria and has agreed to sponsor the lab.

File: labs/blockchain-explorer-iroha.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+Fork of [hyperledger/blockchain-explorer](https://github.com/hyperledger/blockchain-explorer) with support for Hyperledger Iroha.
+
+# Scope of Lab
+Code of [hyperledger/blockchain-explorer](https://github.com/hyperledger/blockchain-explorer) supported only Hyperledger Fabric and needed refactoring.  
+This fork supports only Hyperledger Iroha. It rebuilds backend part from scratch and refactors frontend part.
+
+# Initial Committers
+- https://github.com/turuslan
+- https://github.com/Pobepto
+
```
Comment (arsulegai) [inline labs/blockchain-explorer-iroha.md]: The `Pre-existing repository` section seems to be missing. Could you please add explanation on whether this repository will be used for the development or the code is available in a fork already?

**Comment by lehors** [comment]
Like @trbs , I wonder why this is meant to be done as an incompatible change rather than an addition to Explorer. And I have to ask whether the Explorer team was contacted to see if this couldn't simply be done as part of Explorer rather than a lab.

File: labs/blockchain-explorer-iroha.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Lab Name
+[Integration of Hyperledger Iroha into Hyperledger Explorer tool](https://wiki.hyperledger.org/display/INTERN/Integration+of+Hyperledger+Iroha+into+Hyperledger+Explorer+tool)
+
+# Short Description
+Internship project: [Integration of Hyperledger Iroha into Hyperledger Explorer tool](https://wiki.hyperledger.org/display/INTERN/Integration+of+Hyperledger+Iroha+into+Hyperledger+Explorer+tool).  
+Fork of [hyperledger/blockchain-explorer](https://github.com/hyperledger/blockchain-explorer) with support for Hyperledger Iroha.
+
+# Scope of Lab
+Code of [hyperledger/blockchain-explorer](https://github.com/hyperledger/blockchain-explorer) supported only Hyperledger Fabric and needed refactoring.  
+This fork supports only Hyperledger Iroha. It rebuilds backend part from scratch and refactors frontend part.
```
[the remark this answers is not in the document]
Comment (vipinsun) [reply labs/blockchain-explorer-iroha.md]: I dont get that impression.

**Comment by vipinsun** [comment]
I am not against forking HL-explorer to be an Iroha-explorer, if that is what you intended. However name, as well as sponsor needs to be changed also clear reference to previous repo. Maybe to satisfy the prior reviewers, you can also add rationale of why the Iroha-Explorer has to be labs .

**Comment by nfrunza** [comment]
FYI: HLExplorer was designed to support multiple blockchain platforms, you will need to add your implementation under the https://github.com/hyperledger/blockchain-explorer/tree/master/app/platform, define your DB models, create mapping between UI and DM model if some are not matching the DB endpoints, add starting scripts for your platform.

**Comment by lehors** [comment]
I haven't seen any response to @nfrunza 's comment which actually confirms my doubts about the approach. I would like this proposal to be abandoned and instead the effort to be redirected to contributing to the Explorer project.

**Pull request closed without merging**